polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5032,7 +5099,8 @@ void MLIRASTConsumer::run() {
       if (functionsToEmit.size()) {
         llvm::dbgs() << "-- FUNCTION(S) LEFT TO BE EMITTED --\n";
 
-        for (const auto *FD : functionsToEmit) {
+        for (const auto &F : functionsToEmit) {
```

Comment:
const auto &F -> const FunctionOpInterface &F

Reply:
Yes, it is. `functionsToEmit` contain `FunctionToEmit`s